resources/design-document.md

```diff
@@ -86,30 +86,30 @@ _I will store Medications in a dynamoDbTable. I will store Notifications in a dy
 String customerId;
 String medName;
 String medInfo;
-Set<String> notificationTime;
+Set<Notification> notifications;
 ```
 
 ```
 // NotificationModel
 
 String customerId;
-String time;
 String medName;
-String medInfo;
+String time;
 ```
 
+###Notes - inorder to to populate the medication model fully, the activity calls on the medication dao to get the medName, and med Info, and also calls the notification dao to get the set of notifications and query the notifications table based on the customerId and medName to get all the notifications for that medName. The MedicationModel is used for the health chart.
+
 ## 6.2. _Get All Medications Endpoint_
 
-* Accepts `GET` requests to `/medications/:customerId`
-* Scans medication table based on customerId and returns all MedicationModels for a customerId.
-    * If there are no medications on the table, return an empty Set.
-  
-![Sequence Diagram Get Medication.png](images%2FdesignImages%2FSequence%20Diagram%20Get%20Medication.png)
+* Accepts `GET` requests to `/medications/`
+* Query on just hashkey.  Medication table based on customerId and returns all Medications for a customerId.
+* If there are no medications on the table, return an empty Set.
 
+![Sequence Diagram Get Medications.png](images%2FdesignImages%2FSequence%20Diagram%20Get%20Medications.png)
 
 ## 6.3 _Add Medication Endpoint_
 
-* Accepts `POST` requests to `/medications/:customerId`
+* Accepts `POST` requests to `/medications/:medName`
```

Comment:
The URI for a POST would typically be a collection (e.g., `/medications`), with any specific details being part of the request body.

https://restfulapi.net/rest-put-vs-post/

Reply:
oh ok, so the path is a set of medication objects, but the request body would just be medName?

Reply:
Yes, if that's all you have at that time. (I was thinking you'd also have medInfo.)